src/compiler/tracing.ts

```diff
@@ -145,14 +145,15 @@ namespace ts { // eslint-disable-line one-namespace-per-file
         }
         // sample every 10ms
         const sampleInterval = 1000 * 10;
-        function writeStackEvent(index: number, endTime: number) {
+        function writeStackEvent(index: number, endTime: number, results?: Args) {
             const { phase, name, args, time, separateBeginAndEnd } = eventStack[index];
+            const combinedArgs = { ...args, ...results };
             if (separateBeginAndEnd) {
-                writeEvent("E", phase, name, args, /*extras*/ undefined, endTime);
+                writeEvent("E", phase, name, combinedArgs, /*extras*/ undefined, endTime);

```

Comment:
I suspect, but don't know, that the visualizer dislikes this.  I'm almost certain that `analyze-trace` will silently ignore the new properties.

Reply:
Oh, got it. Sure, I’m happy to just disallow results on separate beginning/end events.